compiler/src/regalloc.ml

```diff
@@ -68,19 +68,19 @@ let types_cannot_conflict reg_size kx x ky y : bool =
   | _, _ -> true
 
 let find_equality_constraints (id: instruction_desc) : arg_position list list =
-  let tbl : (Datatypes.nat, arg_position list) Hashtbl.t = Hashtbl.create 17 in
+  let tbl : (int, arg_position list) Hashtbl.t = Hashtbl.create 17 in
```

Comment:
Out of curiosity, what was the problem with `Datatypes.nat` there?

Reply:
This change does not exist.

Reply:
I know this is not in the final version, but at some point you turn the `int` into `Datatypes.nat` and then back. I was just curious to know why.

Reply:
This is called a mistake.

Reply:
I tought it was a change made on purpose that turned out not to work, if it was just a mistake from the beginning, my question does not make sense indeed.

Reply:
Let me tell you the story of a function whose parameters are called `n` and `p`, in this order, and then a few lines below said function is invoked with arguments `p` and `n`, in this order. Then the hasty reader went lost wondering whether they changed the type of `n` or the type of `p`. (Correct answer is: none, and both).